### PR TITLE
check for a given ns istio install before warn

### LINKED ISSUE #28775: External istiod install generates an confusing warning
**Bug description**

Following up [external istiod doc](https://preliminary.istio.io/latest/docs/setup/additional-setup/external-controlplane/):

```
$ istioctl install -f external-istiod.yaml --set tag=latest --set hub=gcr.io/istio-testing --revision=external
! Istio is being upgraded from 1.8.0 -> 1.9.0.
! Before upgrading, you may wish to 'istioctl analyze' to check for IST0002 deprecation warnings.
This will install the Istio 1.9.0 profile into the cluster. Proceed? (y/N) n
Cancelled.
```

or

```
$ istioctl install -f external-istiod.yaml --set tag=latest --set hub=gcr.io/istio-testing                              
! Istio control planes installed: 1.8.0.
! Use --revision or --force to install Istio.
This will install the Istio 1.9.0 profile into the cluster. Proceed? (y/N) ^C
```

The warning message here is super confusing and should not be necessary because external-istiod.yaml is installing istiod in a different ns from default istio-system so no conflicts.

[ ] Docs
[ x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
Would not expect such a warning when installing to a different ns.

cc @rvennam

### PR DESCRIPTION
**Pull request by linsun**
fixes: #28775
rework of https://github.com/istio/istio/pull/28776

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: operator/cmd/mesh/install.go [hunk 1]
```diff
@@ -337,5 +339,5 @@ func getProfileAndEnabledComponents(setOverlay []string, inFilenames []string, f
 			}
 		}
 	}
-	return profile, enabledComponents, nil
+	return profile, v1alpha12.Namespace(iop.Spec), enabledComponents, nil
```
Comment (irisdingbj) [inline operator/cmd/mesh/install.go:342]: Do we need to change to `controller.IstioNamespace` if v1alpha12.Namespace(iop.Spec) is ""?

Comment (morvencao) [reply operator/cmd/mesh/install.go:342]: +1 to add guard for the empty namespace returned by `v1alpha12.Namespace(iop.Spec)`

Comment (linsun) [reply operator/cmd/mesh/install.go:349]: +1 great comment!!  Fixed

**Review by morvencao — APPROVED** [review]

**Pull request merged**